Add junit fileattribute

bmack123 opened the pull request:
**Is your pull request related to a problem? Please describe.**
Original work completed in #1471, but the owner isn't moving the code forward and I don't have access to the repo, so I forked and added the necessary CHANGELOG.

Original PR Description from #1471:

Allow for adding file= attribute to junit
Use cli arg -f junit,fileattribute=true
Is your pull request related to a problem? Please describe.
CircleCI has a test splitting feature that will split tests across multiple jobs based on the time spent. This feature requires a file attribute to be added. The cucumber-ruby test junit formatter does not include this attribute.

**Describe the solution you have implemented**
This adds a -f junit,fileattribute=true cli option that will add the file attribute to the <testcase/> elements in the junit formatter. By default, all junit formatter output will be unchanged unless this additional option is added.

**Additional context**
This is a continuation of #1471.

olleolleolle commented:
(I guess that the release process is not to be done in this PR. It's a separate process.)

bmack123 commented:
I can just include the CHANGELOG, I wasn't sure from reading the Contributing doc.

bmack123 commented:
I put what I think I would like in a changelog, if there is a different style that you guys want or anything else I need to do, just let me know and I'll take care of it.

luke-hill commented on CHANGELOG.md:
```diff
@@ -24,6 +24,12 @@ Please visit [cucumber/CONTRIBUTING.md](https://github.com/cucumber/cucumber/blo
 
 ### Security fixes
 
+## [NEW VERSION](https://github.com/cucumber/cucumber-ruby/compare/v5.2.0...v5.3.0)
```
Just add this under in git. Then once we cut we just alter the version candidate accordingly.

olleolleolle commented:
@bmack123 Also, 🎉 THANKS  :hearts: :yellow_heart: :green_heart: :blue_heart: :purple_heart: for creating this Pull Request. It's very awesome to see a continuation PR like this one.

bmack123 commented:
Moved the `Added` field under the `IN GIT` section. I removed the excess fields because that felt correct, but if they should stay I can easily add them back in.

bmack123 commented:
Anything else that I can do to move this along and help?

aslakhellesoy commented:
Hi @bmack123,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber
